txn.go

```diff
@@ -334,6 +334,39 @@ func (txn *Txn) First(table, index string, args ...interface{}) (interface{}, er
 	return value, nil
 }
 
+// LongestPrefix is used to fetch the longest prefix match for the given
+// constraints on the index. Note that this will not work with the memdb
+// StringFieldIndex because it adds null terminators which prevent the
```

Comment:
@ryanuber I thought that, too. You can do that to strip the `\x00` off the prefix, but when the `LongestPrefix` routine looks in the tree it hits the node above the actual entry in the index, so it doesn't report anything since that node isn't a leaf. Here's an example:

Say you have `foo` in your tree with a string field indexer. That'll be in there as `foo\x00`. When you do this algorithm for `foobar` it'll scan for the common prefix `foo` just fine and end up at a node with a prefix of `foo` and an edge to `\x00` which is the proper leaf. It doesn't know that it should follow that edge, though, so it just says that there's no entry in there. Because we always put a suffix in there for the string indexer it always runs into this. If you knew to look up `foo\x00` it would work, but you don't know what the longest prefix is when you do a query.